geomstats/discretized_curves_space.py

```diff
@@ -0,0 +1,424 @@
+                                         base_point[:, :-1, :])
+        srv_initial_derivative = gs.transpose(coef_1) * \
+            gs.transpose(tangent_vec_derivative, (2, 1, 0)) + \
+            gs.transpose(coef_2) * gs.transpose(base_point_velocity, (2, 1, 0))
+        srv_initial_derivative = gs.transpose(srv_initial_derivative,
+                                              (2, 1, 0))
+
+        srv_end_point = self.L2_metric.exp(tangent_vec=srv_initial_derivative,
+                                           base_point=srv_base_point)
+        origin_end_point = self.embedding_metric.exp(
+                tangent_vec=tangent_vec[:, 0, :],
+                base_point=base_point[:, 0, :])
+        end_point = self.square_root_velocity_inverse(srv_end_point,
+                                                      origin_end_point)
+
+        return end_point
+
+    def log(self, point, base_point):
+        """
+        Riemannian logarithm of a point wrt a base point.
+        """
+        if not isinstance(self.embedding_metric, EuclideanMetric):
+            raise AssertionError('The logarithm map is only implemented '
+                                 'for dicretized curves embedded in a '
+                                 'Euclidean space.')
+        point_ndim = 2
+        n_sampling_points = point.shape[-2]
+
+        point = gs.to_ndarray(point, to_ndim=point_ndim + 1)
+        base_point = gs.to_ndarray(base_point, to_ndim=point_ndim + 1)
+        n_points = point.shape[0]
+
+        srv_point = self.square_root_velocity(point)
+        srv_base_point = self.square_root_velocity(base_point)
+        base_point_velocity = n_sampling_points * (
+                base_point[:, 1:, :] - base_point[:, :-1, :])
+        base_point_velocity_norm = self.pointwise_norm(base_point_velocity,
+                                                       base_point[:, :-1, :])
+
+        coef_1 = gs.sqrt(base_point_velocity_norm)
+        coef_2 = 1 / base_point_velocity_norm ** (3/2) * \
+            self.pointwise_inner_product(srv_point - srv_base_point,
+                                         base_point_velocity,
+                                         base_point[:, :-1, :])
+        log_derivative = gs.transpose(coef_1) * \
+            gs.transpose(srv_point - srv_base_point, (2, 1, 0)) + \
+            gs.transpose(coef_2) * gs.transpose(base_point_velocity, (2, 1, 0))
+        log_derivative = gs.transpose(log_derivative, (2, 1, 0))
+        log_0 = self.embedding_metric.log(point=point[:, 0, :],
+                                          base_point=base_point[:, 0, :])
+        log_0 = gs.transpose(np.tile(log_0, (1, 1, 1)), (1, 0, 2))
+
+        log_cumsum = gs.hstack([gs.zeros((n_points, 1, 3)),
+                                np.cumsum(log_derivative, -2)])
+        log = log_0 + 1 / n_sampling_points * log_cumsum
+
+        return log
+
+    def geodesic(self, initial_point,
```

Comment:
Same: can this function call the parent's function `geodesic` to avoid copy-pasting code?